client/src/components/views/Orders.jsx

```diff
@@ -0,0 +1,236 @@
+import React, { Fragment, useState, useRef } from 'react';
+import {
+  Center,
+  useToast,
+  Textarea,
+  Select,
+  useDisclosure,
+  Table,
+  Thead,
+  Tbody,
+  Tr,
+  TableCaption,
+  Heading,
+  Button,
+  FormControl,
+  FormLabel,
+  Input,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+} from '@chakra-ui/react';
+import { SmallAddIcon } from '@chakra-ui/icons';
+import Loader from 'components/common/Loader.jsx';
+import { getOrders, getMaterialList, postOrders } from 'utils/api/orders.js';
+import { useQuery } from 'react-query';
+import { StyledTableRow, StyledTableHeader, StyledTableCell } from 'components/common/Table.jsx';
+import { TablePagination } from '@material-ui/core';
+import { NoResultImage } from 'components/common/Image.jsx';
+
+const Orders = () => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const { isLoading, isSuccess, data, refetch } = useQuery('orders', getOrders);
+  const [page, setPage] = useState(0);
+  const [rowsPerPage, setRowsPerPage] = useState(10);
+  const toast = useToast();
+  const [material, setMaterial] = useState();
+  const [cost, setCost] = useState(0);
+  const [deliveryDate, setDeliveryDate] = useState();
+  const [orderDate, setOrderDate] = useState();
+  const [quantity, setQuantity] = useState(0);
+  const manufacturer = useRef('');
+  const location = useRef('');
+  const note = useRef('');
+  const materialTypes = ['rubber', 'aluminum', 'steel', 'copper', 'plastic', 'leather'];
+  const materialCost = useQuery('orders/materialList', getMaterialList);
+
+  const handleChangePage = (event, newPage) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
+
+  // sets material and cost states
+  const handleMaterial = (e) => {
+    const choice = e.target.value;
+    setMaterial(choice);
+    setCost(materialCost.data.data[choice].cost);
+  };
+
+  // generates a random int within ranges
+  function getRandomInt(min, max) {
+    return Math.floor(Math.random() * (max - min)) + min;
+  }
+
+  const handleSubmit = async () => {
+    const orderTime = new Date();
+    const deliveryTime = new Date();
+    const randomDeliveryDate = deliveryTime.getDate() + getRandomInt(1, 8); // random delivery date within a week
+    deliveryTime.setDate(randomDeliveryDate);
+
```

Comment:
The date on the website is showing in a weird format
image
![image](https://user-images.githubusercontent.com/37958655/111553014-e3e1d780-8759-11eb-874e-c0b4801d9926.png)

I use the format: today.getFullYear() + '-' + (today.getMonth() + 1) + '-' + today.getDate() + ' ' + today.getHours() + ':' + today.getMinutes() + ':' + today.getSeconds()

Reply:
cool, will add that

Reply:
This might be simpler to use than manipulating the string manually.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toTimeString

There are other `.to___String()` methods that could be useful or better on the MDN docs.